kedro-datasets/kedro_datasets/pandas/deltatable_dataset.py

```diff
@@ -0,0 +1,258 @@
+"""``DeltaTableDataSet`` loads/saves delta tables from/to a filesystem (e.g.: local,
+S3, GCS), Databricks unity catalog and AWS Glue catalog respectively. It handles
+load and save using a pandas dataframe.
+"""
+from copy import deepcopy
+from typing import Any, Dict, List, Optional
+
+import pandas as pd
+from deltalake import DataCatalog, DeltaTable, Metadata
+from deltalake.exceptions import TableNotFoundError
+from deltalake.writer import write_deltalake
+from kedro.io.core import AbstractDataSet, DataSetError
+
+
+class DeltaTableDataSet(AbstractDataSet):  # pylint:disable=too-many-instance-attributes
```

Comment:
Any chance we can make this a versioned dataset? Is the `DeltaTable` version compatible with Kedro's own timestamp versioning or it has to be a incremental version i.e. 1,2,3 ?

Reply:
`DeltaTable` can accept both - incremental version numbers i.e. 1, 2, 3 or a `datetime` string as stated here: https://delta-io.github.io/delta-rs/python/usage.html#time-travel

If we make this a versioned dataset, we would be adding `version` to accepted list of attributes in `__init__`. Which `version` attribute would users refer to if they want to load a specific version of `DeltaTable` from Kedro `yml` catalog?  Would that confuse users with `DeltaTable` version?

Reply:
IMHO, If we allow it to be also a versioned dataset, we are allowing the user to do a nested versioning structure where the Delta Table is on top of the Kedro version. This may not be useful practically. I think if user want to use Delta Table, they want to leverage only the versioning in Delta table. Adding one more versioning layer to it seems redundant and can create confusion.

Reply:
I agree, let's keep this as is. Delta's versioning is more powerful than Kedro's one. I will leave the discussion in the dedicated issue. It does creates some inconsistent behavior, for example you can't use the `--load-version` CLI argument if you use Delta, but I guess this is not too important.

- https://github.com/kedro-org/kedro/issues/2703